### DISCUSSION TITLE
How are wezterm's workspace's compared to tmux's sessions

### DISCUSSION THREAD

**Discussion by Tanish2002** [wezterm/wezterm#5061] [General]
Hey guys. I recently tried out wezterm and liked it a lot. 
I was using tmux alongside suckless st before. Now I'm thinking of switching my setup over to wezterm.
But I'm confused about workspaces in wezterm. Wezterm has first class support for multiplexing which is one reasons why I use tmux.

The second reason, why I use tmux is for sessions. I have configured [tmux-resurrect](https://github.com/tmux-plugins/tmux-resurrect) as plugin so I can have persistent sessions. Which is pretty nice for my setup since I can simply attach to my old session and continue my work. 

Is this setup possible by using wezterm's workspaces? From what I've read, workspaces can be used to define the panes and applications that need to opened. But I don't want that. I simply want to open a "workspace" open my windows, run neovim inside one window, some dev server in other and then I can simply detach and reattach and find all my apps running. And this should be persistent over restarts.

PS: I know that tmux can be used with wezterm but from what I've seen wezterm can replicate most of what I do tmux, so If possible I would like to just remove tmux and use wezterm.

**Comment by MartyMcFlyInTheSky** [comment]
I have found this article, maybe it helps you: https://www.florianbellmann.com/blog/switch-from-tmux-to-wezterm#introduction

In the meantime, I am myself figuring out how to use wezterm multiplexing with our production servers instead of tmux.

**Comment by fredrikaverpil** [comment]
I also wrote up my impressions on wezterm workspaces: https://fredrikaverpil.github.io/blog/2024/10/20/session-management-in-wezterm-without-tmux/
Hint: they're awesome together with fuzzy searching!
